src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -0,0 +1,77 @@
+package com.google.sps.servlets;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.YouTube;
+import java.io.IOException;
+import java.lang.Exception;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+
+public class CommentService {
+
+  private final String DEVELOPER_KEY = "DEV_KEY";
+  private static final String APPLICATION_NAME = "YouTube Comments";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private YouTube getService() throws GeneralSecurityException, IOException{
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
+        .setApplicationName(APPLICATION_NAME)
+        .build();
+  }
+
+  /**
+   * Calls YouTube Data API to return a list of the top 25 comments for a video, identified by its
+   * video id. 
+   */
+  public List<String> getCommentsFromId(String videoId)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
```

Comment:
Nit: This method builds a service client every time it is called. Is it possible to only do this once? Not even sure if its useful to do so.

Reply:
ack, I tried to make a CommentService constructor that would call getService, but then whenever you create a CommentService object you'd have to throw or catch GeneralSecurityException and IOException. Doable but I'm not sure if it's worth it?

Reply:
Hmm.. What happens if instead of inside a constructor, you fill the object property when you declare it. I'm guessing you would have the same issue but interested if it didn't

Reply:
I think the same issue would apply, but I ended up using a constructor anyway haha